Makefile

```diff
@@ -20,6 +20,14 @@ test: generate
 integration-test: generate
 	go test -tags=integration ./...
 
+# Run the end to end tests and requires GITHUB_AUTH_TOKEN to be set. If not the tests will fail.
+# Not included in integration tests because it requires a github token.
+# To run the tests locally, run `GITHUB_AUTH_TOKEN=<your token> make e2e-test`
+# https://github.com/ossf/scorecard#authentication
+.PHONY: e2e-test
+e2e-test: generate
+	go test -tags=e2e ./...
```

Comment:
it would be nice to have a check for token here and if it is not set to echo that GITHUB_AUTH_TOKEN needs to be set.

Reply:
Good idea, I have added a check for `GITHUB_AUTH_TOKEN`:

```
check-env:
ifndef GITHUB_AUTH_TOKEN
	$(error GITHUB_AUTH_TOKEN is not set)
endif
```